ui/src/app/base/components/VLANSelect/VLANSelect.tsx

```diff
@@ -15,26 +18,44 @@ type Props = {
   defaultOption?: { label: string; value: string } | null;
   fabric?: VLAN["fabric"];
   includeDefaultVlan?: boolean;
+  showSpinnerOnLoad?: boolean;
+  setDefaultValueFromFabric?: boolean;
   vlans?: VLAN[] | null;
 } & FormikFieldProps;
 
 export const VLANSelect = ({
   defaultOption = { label: "Select VLAN", value: "" },
   fabric,
   includeDefaultVlan = true,
+  showSpinnerOnLoad = false,
+  setDefaultValueFromFabric,
   name,
   vlans,
+  disabled,
   ...props
 }: Props): JSX.Element => {
   const dispatch = useDispatch();
   let vlanList: VLAN[] = useSelector(vlanSelectors.all);
   const vlansLoaded = useSelector(vlanSelectors.loaded);
+  const selectedFabric = useSelector((state: RootState) =>
+    fabricSelectors.getById(state, fabric)
+  );
+  const { setFieldValue } = useFormikContext();
+
+  useEffect(() => {
+    if (setDefaultValueFromFabric) {
+      const vlan = selectedFabric?.default_vlan_id;
+      if (vlan) {
```

Comment:
Usually I try to avoid simple truthy checks like this when dealing with IDs, since some model IDs are indexed at 0. For VLANs this isn't the case (I think it starts at 5001 to completely avoid overlapping with the `vid`) but it's good to keep in mind.

Reply:
I recently added a util you could use here: `isId` that supports checking for `0` so we don't get false negatives: https://github.com/canonical-web-and-design/maas-ui/blob/master/ui/src/app/utils/isId.ts